includes/functions.php

```diff
@@ -1593,3 +1593,143 @@ function is_self_ping( $id ) {
 
 	return false;
 }
+
+/**
+ * Get the embed of a URL.
+ *
+ * This function supports both ActivityPub JSON and oEmbed URLs.
+ * It first tries to fetch ActivityPub JSON, and if that fails,
+ * falls back to oEmbed.
+ *
+ * @param string $url The URL to get the embed of.
+ *
+ * @return string|false The embed of the URL or false if not found.
+ */
+function embed_get( $url ) {
+	$embed = \wp_oembed_get( $url );
+	if ( $embed ) {
+		return $embed;
+	}
+
+	// Create unique transient keys for both HTTP calls.
+	$main_transient_key = '_activitypub_embed_' . md5( $url );
+	$embed              = \get_transient( $main_transient_key );
+
+	if ( false === $embed ) {
+		$response = Http::get( $url );
```

Comment:
There is already a function that provides all the needed headers and has caching built in: https://github.com/Automattic/wordpress-activitypub/blob/b2e14e9e00350d2cadfef1f08e0241a55f637610/includes/class-http.php#L224C25-L224C42